aptos-move/framework/aptos-framework/sources/stake.spec.move

```diff
@@ -0,0 +1,168 @@
+spec aptos_framework::stake {
+
+    // -----------------
+    // Global invariants
+    // -----------------
+
+    spec module {
+        // The validator set should satisfy its desired invariant.
+        invariant [suspendable] validator_set_is_valid();
```

Comment:
What does [suspendable] do?

Reply:
`[suspendable]` is related to the semantics of global invariant evaluation. By default, invariants are checked immediately at the point where an update on global memory happens (with borrow semantics, this is the time when a `&mut` for a global borrow goes out of scope). However, there are situations where this is not practical. For invariants marked as 'suspendable' evaluation is allowed to be postponed to a later point, for example, after a function has terminated. I'm not seeing where Junkil is actually using this; these declarations just _allow_ to do it.

I'm seeing that [the documentation](https://github.com/move-language/move/blob/main/language/move-prover/doc/user/spec-lang.md) does not mention those features, which has been lately added by David Dill. This is something which we desperately need to fix before we all forget about ;-)  @junkil-park should we open a github issue for this?

Reply:
`[suspendable]` is needed for `on_new_epoch`, which has `pragma disable_invariants_in_body`. I created a github issue for it: https://github.com/move-language/move/issues/477